src/runtime/pythonengine.cs

```diff
@@ -538,6 +546,56 @@ public static GILState GIL()
 
             return new GILState();
         }
+        
+        private static Dictionary<string, PyScope> NamedScopes = new Dictionary<string, PyScope>();
+
+        public static PyScope CreateScope()
+        {
+            return CreateScope(null);
+        }
+
+        public static PyScope CreateScope(string name)
```

Comment:
By contrast, I want to split the two methods CreateScope() and CreateScope(string name).

CreateScope() is unique, CreateScope(string name) will be put in the ScopeManager class.